### ISSUE TITLE
std::bad_alloc error

### ISSUE THREAD

**Issue by RubD** [GuipengLi/ChIA-PET2#15]
Hi,

The pipeline always throws an error at the 'buildBedpe' step:

terminate called after throwing an instance of ‘std::bad_alloc’
 what():  std::bad_alloc
Aborted (core dumped)

I'm not sure why this is. I tried 3 different servers and all of them should have enough available memory.

Thanks,
Ruben

**Comment by GuipengLi** [comment]
What's the command you used?

**Comment by RubD** [comment]
This is the command I used:
`ChIA-PET2 --genome $bwa_genome_index --bedtoolsgenome $bedtools_genome --linkerA CGCGATATCTATCTG --linkerB ACGCGATATCTTATC --output $output_dir --name shLUC_subtest --mode 1 --err 1 --keepempty 2 --thread 10 --forward $forward_read --reverse $reverse_read`

Initially I got an error at the buildBedpe stage about a problem with the names in the .sam files. So I first ran `samtools sort -n file_1_R.sam > file_1_R.nsorted.sam` for all .sam files.

Then I ran the buildBedpe command `buildBedpe file_1_R.nsorted.sam file_2_R.nsorted.sam 30 10 1` or the ChIA-PET2 command starting from point 3 with both of them ending with the same error.
